### PR TITLE
feat: add vs ai bot template

### PR DESCRIPTION
**Pull request by xiaolang124**
feat: https://msazure.visualstudio.com/Microsoft%20Teams%20Extensibility/_workitems/edit/25123363/

@microsoft/teamsfx-cli	v2.1.0-rc.0
@microsoft/teamsfx-core	v2.1.0-rc.0

Fx-core feat commits:
 feat: add vs ai bot template 7fd7c9a3a 

CLI feat commits:
  

Extension-toolkit feat commits:
  

SDK feat commits:
  

SDK React feat commits:
  

.Net SDK feat commits:
  

Fx-core fix commits:
  

CLI fix commits:
  

Extension-toolkit fix commits:
  

SDK fix commits:
  

SDK React fix commits:
  

.Net SDK fix commits:

### CODE REVIEW

File: templates/csharp/ai-bot/Controllers/BotController.cs.tpl [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    private readonly CloudAdapter Adapter;
+    private readonly IBot Bot;
+
+    public BotController(CloudAdapter adapter, IBot bot)
+    {
+        Adapter = adapter;
+        Bot = bot;
+    }
+
+    [HttpPost, HttpGet]
```
Comment (swatDong) [inline templates/csharp/ai-bot/Controllers/BotController.cs.tpl:23]: could `HttpGet` be removed?

**Review by swatDong — APPROVED** [review]

**Review by kuojianlu — APPROVED** [review]

File: templates/csharp/ai-bot/GettingStarted.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+1. Set your [OpenAI API Key](https://openai.com/api/) to *appsettings.Development.json*.
+
+    ```json
+      "OpenAI": {
+        "ApiKey": "<your-openai-api-key>"
+      },
+    ```
+1. In the debug dropdown menu, select Dev Tunnels > Create A Tunnel (set authentication type to Public) or select an existing public dev tunnel
+1. Right-click your project and select Teams Toolkit > Prepare Teams App Dependencies
```
Comment (sffamily) [inline templates/csharp/ai-bot/GettingStarted.md:24]: Please check and add "." after each bullet/sentence.

File: templates/csharp/ai-bot/GettingStarted.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+> - [Azure OpenAI](https://aka.ms/oai/access) resource or an account with [OpenAI](https://platform.openai.com).
+
+1. Set your [OpenAI API Key](https://openai.com/api/) to *appsettings.Development.json*.
+
+    ```json
+      "OpenAI": {
+        "ApiKey": "<your-openai-api-key>"
+      },
+    ```
+1. In the debug dropdown menu, select Dev Tunnels > Create A Tunnel (set authentication type to Public) or select an existing public dev tunnel
```
Comment (sffamily) [inline templates/csharp/ai-bot/GettingStarted.md:23]: recommend to make it:
.... > Create A Tunnel, set authentication type to Public, or .....

File: templates/csharp/ai-bot/Controllers/BotController.cs.tpl (hunk 1 above)
Comment (xiaolang124) [reply templates/csharp/ai-bot/Controllers/BotController.cs.tpl]: Done.

File: templates/csharp/ai-bot/GettingStarted.md (hunk 2 above)
Comment (xiaolang124) [reply templates/csharp/ai-bot/GettingStarted.md]: Fixed.

File: templates/csharp/ai-bot/GettingStarted.md (hunk 3 above)
Comment (xiaolang124) [reply templates/csharp/ai-bot/GettingStarted.md]: Done.

**Review by swatDong — DISMISSED** [review]

**Comment by xiaolang124** [comment]
feat: https://msazure.visualstudio.com/Microsoft%20Teams%20Extensibility/_workitems/edit/25123363/

**Review by swatDong — APPROVED** [review]

**Review by jayzhang — APPROVED** [review]

**Review by nliu-ms — APPROVED** [review]

**Review by Alive-Fish — APPROVED** [review]

**Review by LongOddCode — APPROVED** [review]

**Pull request merged**